panic: VERIFY(vm_page_none_valid(ma[i])) failed

<!-- Please fill out the following template, which will help other contributors address your issue. -->

<!--
Thank you for reporting an issue.

*IMPORTANT* - Please check our issue tracker before opening a new issue.
Additional valuable information can be found in the OpenZFS documentation
and mailing list archives.

Please fill in as much of the template as possible.
-->

### System information
<!--  add version after "|" character -->
Type | Version/Name
 --- | ---
Distribution Name	| FreeBSD current
Distribution Version	|
Kernel Version	| main-bde575b273eeb
Architecture	| amd64
OpenZFS Version	| zfs-2.3.99-64-FreeBSD_g1c9a4c8cb zfs-kmod-2.3.99-92-FreeBSD_gd0a91b9f8

<!--
Command to find OpenZFS version:
zfs version

Commands to find kernel version:
uname -r		# Linux
freebsd-version -r	# FreeBSD
-->

### Describe the problem you're observing
Stress test with a KASAN kernel build

### Describe how to reproduce the problem
cd /usr/src/tools/test/stress2/misc; ./all.sh -o zfs11.sh
### Include any warning/errors/backtraces from the system logs
<!--
*IMPORTANT* - Please mark logs and text output from terminal commands
or else Github will not display them correctly.
An example is provided below.

Example:
```
this is an example how log text should be marked (wrap it with ```)
```
-->

```
20241126 01:32:46 all (64/676): zfs11.sh
ZFS filesystem version: 5
ZFS storage pool version: features support (5000)
panic: VERIFY(vm_page_none_valid(ma[i])) failed

cpuid = 4
time = 1732581173
KDB: stack backtrace:
db_trace_self_wrapper() at db_trace_self_wrapper+0xa5/frame 0xfffffe014d3201f0
kdb_backtrace() at kdb_backtrace+0xc6/frame 0xfffffe014d320350
vpanic() at vpanic+0x214/frame 0xfffffe014d3204f0
spl_panic() at spl_panic+0xa9/frame 0xfffffe014d3205b0
zfs_freebsd_getpages() at zfs_freebsd_getpages+0x36a/frame 0xfffffe014d320740
VOP_GETPAGES_APV() at VOP_GETPAGES_APV+0x91/frame 0xfffffe014d320770
vnode_pager_getpages() at vnode_pager_getpages+0x12d/frame 0xfffffe014d320870
vm_pager_get_pages() at vm_pager_get_pages+0xae/frame 0xfffffe014d3208d0
vm_fault() at vm_fault+0xedf/frame 0xfffffe014d320be0
vm_fault_trap() at vm_fault_trap+0x95/frame 0xfffffe014d320c30
trap_pfault() at trap_pfault+0x3a6/frame 0xfffffe014d320d50
trap() at trap+0x817/frame 0xfffffe014d320f30
calltrap() at calltrap+0x8/frame 0xfffffe014d320f30
--- trap 0xc, rip = 0xdebcd7291cc, rsp = 0xdf3ee63a9f0, rbp = 0xdf3ee63aa50 ---
KDB: enter: panic
[ thread pid 47272 tid 101315 ]
Stopped at      kdb_enter+0x34: movq    $0,0x20f2991(%rip)
db> x/s version
version:        FreeBSD 15.0-CURRENT #0 main-n273839-bde575b273eeb-dirty: Mon Nov 25 21:09:13 CET 2024\012    pho@mercat1.netperf.freebsd.org:/usr/src/sys/amd64/compile/PHO-KASAN\012
db> 
``

--- a/module/os/freebsd/zfs/zfs_vnops_os.c
+++ b/module/os/freebsd/zfs/zfs_vnops_os.c
@@ -3960,10 +3960,8 @@ zfs_getpages(struct vnode *vp, vm_page_t *ma, int count, int *rbehind,
 			 * to the page fault handler's OOM logic, but this is
 			 * the best we can do for now.
 			 */
-			for (int i = 0; i < count; i++) {
-				ASSERT(vm_page_none_valid(ma[i]));
+			for (int i = 0; i < count; i++)
 				vm_page_xunbusy(ma[i]);
-			}
 
 			lr = zfs_rangelock_enter(&zp->z_rangelock,
 			    rounddown(start, blksz), len, RL_READER);

